casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -202,12 +203,15 @@ object Validate {
     import cats.instances.list._
 
     for {
-      parentsPresent <- ProtoUtil.parentHashes(block).forallM(p => dag.contains(p))
-      justificationsPresent <- block.justifications
-                                .forallM(j => dag.contains(j.latestBlockHash))
-      result = if (parentsPresent && justificationsPresent) {
+      missing <- ProtoUtil
+                  .dependenciesHashesOf(block)
+                  .filterA(
+                    d => (dag.contains(d) ||^ dag.equivocationHashes.contains(d).pure[F]).not
```

Comment:
Oh, I guess I'm doing it wrong here... this way block will use equivocation as a valid dependency

Reply:
Fix https://github.com/rchain/rchain/pull/3175/commits/790e510cff557496ca4f2edfe302333655ff9c55